frontend/src/components/AddProduct.vue

```diff
@@ -0,0 +1,540 @@
+                msgError = 'Please enter a Quantity'
+                return msgError
+            }
+
+            else if (formQuantity > 999999) {
+                msgError = "You couldn't possibly have that many, could you?"
+                return msgError
+            }
+
+            else if (formQuantity < 0) {
+                msgError = "Let's try something positive"
+                return msgError
+            }
+
+            else {
+                return false
+            }
+        },
+        
+        checkErrorPrice() {
+            let formPrice = this.$refs.addProductFormPrice.value
+            let msgError = ''
+
+            if (formPrice == '') {
+                msgError = 'Please enter a Price'
+                return msgError
+            }
+            
+            else if (formPrice > 999999) {
+                msgError = "That's a little expensive for our taste!"
+                return msgError
+            }
+
+            else if (formPrice < 0) {
+                msgError = "Rather than pay people to take your product, why not donate it instead?"
+                return msgError
+            }
+
+            else {
+                return false
+            }
+        },
+
+        checkErrorShelfLife(formShelfLife) {
+            let msgError = ''
+
+            if (formShelfLife == '') {
+                msgError = 'Please provide a Shelf Life for the new product'
+                return msgError
+            }
+
+            else if (!this.isAlphaNumeric(formShelfLife)) {
+                msgError = 'Please use letters and numbers only'
+                return msgError
+            }
+
+            else if (formShelfLife.length < 3 || formShelfLife.length > 128) {
+                msgError = 'Shelf Life must be 3 to 128 characters in length'
+                return msgError
+            }
+
+            else {
+                return false
+            }
+        },
+        
+
+        //Change bool form values true/false
+        changeIsPerishable(tf) {
+            this.isPerishable = tf
+        },
+
+        changeKeepRoom(tf) {
+            this.keepRoom = tf
+        },
+
+        changeKeepFridge(tf) {
+            this.keepFridge = tf
+        },
+
+        changeKeepFreezer(tf) {
+            this.keepFreezer = tf
+        },
+
+        //Validate
+        anyErrors() {
+            if(this.hasErrorName == true ||
+            this.hasErrorDescription == true ||
+            this.hasErrorCategory == true ||
+            this.hasErrorQuantity == true ||
+            this.hasErrorPrice == true ||
+            this.hasErrorShelfLifeRoom == true ||
+            this.hasErrorShelfLifeFridge == true ||
+            this.hasErrorShelfLifeFreezer == true) {
+                return true
+            }
+            else {
+                return false
+            }
+        },
+
+        validateForm() {
+            document.getElementById('errorName').innerHTML = ''
```

Comment:
I'm going to go ahead and let this one slide for the purposes of MVP, but I'm not a huge fan of using JS query selectors in Vue components since they often times don't get the true value.